src/lib.rs

```diff
@@ -162,3 +219,206 @@ impl LevelIndex {
         }
     }
 }
+
+bitflags::bitflags! {
+    #[repr(transparent)]
+    pub struct ChannelTypeQualifiers: u32 {
+        const LINEAR        = (1 << 0);
+        const EXPONENT      = (1 << 1);
+        const SIGNED        = (1 << 2);
+        const FLOAT         = (1 << 3);
+    }
+}
+
+bitflags::bitflags! {
+    #[derive(Default)]
+    #[repr(transparent)]
+    pub struct DataFormatFlags: u32 {
+        const STRAIGHT_ALPHA             = 0;
+        const ALPHA_PREMULTIPLIED        = (1 << 0);
+    }
+}
+
+// This can be obtained from std::mem::transmute::<f32, u32>(1.0f32) but we
+// want to support nostd. It is used for identifying normalized sample types
+// as in Unorm or Snorm
+const F32_1_AS_U32: u32 = 1065353216;
+
+#[derive(Debug, Default)]
+pub struct BasicDataFormatDescriptor<'data> {
+    pub vendor_id: u32,             //: 17;
+    pub descriptor_type: u32,       //: 15;
+    pub version_number: u32,        //: 16;
+    pub descriptor_block_size: u32, //: 16;
+    /// None means Unspecified or is an otherwise unknown value
+    pub color_model: Option<ColorModel>, //: 8;
+    /// None means Unspecified or is an otherwise unknown value
+    pub color_primaries: Option<ColorPrimaries>, //: 8;
+    /// None means Unspecified or is an otherwise unknown value
+    pub transfer_function: Option<TransferFunction>, //: 8;
+    pub flags: DataFormatFlags,     //: 8;
+    pub texel_block_dimensions: [u32; 4], //: 8 x 4;
+    pub bytes_planes: [u32; 8],     //: 8 x 8;
+    sample_data: &'data [u8],
+}
+
+impl<'data> BasicDataFormatDescriptor<'data> {
+    pub fn from_bytes_and_offset(bytes: &'data [u8], initial_offset: usize) -> Self {
```

Comment:
Will downstream code ever need to call this directly? If not, it shouldn't be `pub`.